Sources/ClientRuntime/Networking/Http/HttpResponseBinding.swift

```diff
@@ -4,5 +4,9 @@
  */
 
 public protocol HttpResponseBinding {
-    init(httpResponse: HttpResponse, decoder: ResponseDecoder?) throws
+    init(httpResponse: HttpResponse, decoder: ResponseDecoder?) async throws
```

Comment:
Note that this interface is now async.  (I made the error interface async, figured I'd convert the "happy path" too.)